common/src/main/java/com/hibiscusmc/hmccosmetics/HMCCosmeticsPlugin.java

```diff
@@ -293,4 +279,31 @@ public static boolean isOnLatestVersion() {
     public static String getLatestVersion() {
         return latestVersion;
     }
+
+    private static void loadEmotes() {
+        File emoteDir = new File(getInstance().getDataFolder().getPath() + "/emotes/");
+        if (!emoteDir.exists()) return;
+
+        File[] emoteFiles = emoteDir.listFiles();
+        if (emoteFiles == null || emoteFiles.length == 0) return;
+
+        emoteFiles = Arrays.stream(emoteFiles).filter(file -> file.getPath().endsWith(".bbmodel")).distinct().toArray(File[]::new);
+        if (emoteFiles.length == 0) return;
+
+        for (File animationFile : emoteFiles) {
+            String animationKey = FilenameUtils.removeExtension(animationFile.getName());
+            PlayerAnimator.api.getAnimationManager().importAnimations(animationKey, animationFile);
+        }
+
+        for (Map.Entry<String, AnimationPack> packEntry : PlayerAnimator.api.getAnimationManager().getRegistry().entrySet()) {
+            packEntry.getValue().getAnimations().keySet().forEach(animation -> {
+                String key = packEntry.getKey().replace(":", ".") + "." + animation;
+                emoteMap.put(animation, key);
+            });
+        }
+    }
+
+    public static Map<String, String> getEmoteMap() {
```

Comment:
Maps shouldn't be directly accessed by other classes, methods should control what happens in the Map.

Reply:
Will fix!